memory/provider.go

```diff
@@ -92,3 +96,18 @@ func (d memoryDBProvider) DropDatabase(ctx *sql.Context, name string) (err error
 	delete(d.dbs, strings.ToLower(name))
 	return
 }
+
+// Function implements sql.FunctionProvider
+func (_ memoryDBProvider) Function(_ *sql.Context, name string) (sql.Function, error) {
```

Comment:
Could just omit this

Reply:
Good catch; that's not needed now that the TableFunction method was split out of the FunctionProvider interface.